include/imex/Conversion/GPUXToLLVM/GPUXToLLVMPass.h

```diff
@@ -0,0 +1,52 @@
+//===- GPUXToLLVMPass.h - GPUXToLLVM conversion  ---------------*- C++ -*-===//
+//
+// Copyright 2022 Intel Corporation
+// Part of the IMEX Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+///
+/// \file
+/// This file defines the GPUXToLLVM conversion,
+///
+//===----------------------------------------------------------------------===//
+
+#ifndef IMEX_GPUXTOLLVM_PASS_H_
+#define IMEX_GPUXTOLLVM_PASS_H_
+
+#include <memory>
+#include <mlir/IR/PatternMatch.h>
+#include <mlir/Transforms/DialectConversion.h>
+namespace mlir {
+class ConversionTarget;
+class LLVMTypeConverter;
+class Location;
+struct LogicalResult;
+class Pass;
+class Operation;
+class RewritePatternSet;
+class ModuleOp;
+template <typename T> class OperationPass;
+
+} // namespace mlir
+
+namespace imex {
+
+/// Populates the given list with patterns that convert from GPUX dialect to
+/// LLVM.
+void populateGpuxToLLVMPatternsAndLegality(mlir::LLVMTypeConverter &converter,
+                                           mlir::RewritePatternSet &patterns,
+                                           mlir::ConversionTarget &target);
```

Comment:
A little bit difficult for newbies  to understand the functionality of the API. A description based on parameter names would help.

Reply:
this function here follows similar functionality as other general MLIR Dialect conversion flow specified here - https://mlir.llvm.org/docs/DialectConversion/, I added some description to highlight IMEX specific conversion we added in this function.